### ISSUE TITLE
EntityID are limited to valid URI while external vendors like Okta do not impose this limit

### ISSUE THREAD

**Issue by lems3** [simplesamlphp/simplesamlphp#1891]
**Specifics of your environment**
1. Are you acting as SP/IdP/proxy? 
We're using Okta as an IdP
3. SimpleSAMLphp: What version are you using?
2.0-beta99, but looking at tags for the commit it seems to be available in 2.1 too
5. PHP: What version are you using?
8.1
7. Platform: unix or Windows?
Unix
9. Webserver: Apache/Nginx/ISS?
Apache

**Describe the bug**
When upgrading our SimpleSAML installation, it stopped working. We got this error : 
``` Noticed exception 'SimpleSAML\Assert\AssertionFailedException' with message ''Drupal - site.example.com' is not a valid RFC3986 compliant URI' in /var/www/html/vendor/simplesamlphp/assert/src/Assert.php:363 ```

Digging into this, this seems to be linked to this change :
https://github.com/simplesamlphp/simplesamlphp/pull/1658

**Expected behavior**
EntityID didn't had this restriction before, and audiences in Okta don't have this limitation. Adding this new restriction now seems to be an important breaking change, and would lead to us having to reconfigure many elements. My main issue with this is how other SAML integration don,t have this restriction...

Is this behavior really expected from entityID in SAML? Okta's support page reccomend this, but does not seems to discuss any hard restriction over this : https://support.okta.com/help/s/article/What-Is-the-Audience-URI?language=en_US

**Comment by tvdijen** [comment]
> Is this behavior really expected from entityID in SAML?

Yes, it's specified in the SAML2 specifications and also in the SAML2 XSD-schema's.

**Comment by lems3** [comment]
> > Is this behavior really expected from entityID in SAML?
> 
> Yes, it's specified in the SAML2 specifications and also in the SAML2 XSD-schema's.

Yeah, while digging further into this I realized that it's in the spec now, but clearly a lot of tools were not implementing it properly.

As SimpleSAMLPHP didn't enfore it in 1.x, it could be useful to add a note in the [upgrade notes](https://simplesamlphp.org/docs/devel/simplesamlphp-upgrade-notes-2.0) about this.

Would it be accepted to add a parameter to make this check optional, in case an IdP isn't enforcing this on their site? Right now, on our side we're evaluating the impact of reconfiguring a couple of hundreds of Okta apps... While I'll always support properly enforcing standards, in this situation we'd prefer to have a gradual transition. IMO, the parameter should downgrade the error to a warning.

If not, then this issue can be closed, and we'll either make a local patch for now that handles this issue until we've went trough our apps and reconfigured all audiences.

Thanks!
